File: fsharp/wonderland-number/test/finder-test.fs
module finder_test

open Expecto
open wonderland_number.finder

let hasAllTheSameDigits (n1 : int) (n2 : int) : bool =
    let s1 = n1 |> string |> Set.ofSeq
    let s2 = n2 |> string |> Set.ofSeq
    s1 = s2

[<Tests>]
let tests =
    testList "finder" [
        testCase "A wonderland number must have the following things true about it" <| fun _ ->
            let wondernum = wonderland_number()
            Expect.equal (wondernum |> string |> String.length) 6 ""
            Expect.isTrue (hasAllTheSameDigits wondernum (wondernum * 2)) ""
            Expect.isTrue (hasAllTheSameDigits wondernum (wondernum * 3)) ""
            Expect.isTrue (hasAllTheSameDigits wondernum (wondernum * 4)) ""
            Expect.isTrue (hasAllTheSameDigits wondernum (wondernum * 5)) ""
            Expect.isTrue (hasAllTheSameDigits wondernum (wondernum * 6)) ""
    ]

[<EntryPoint>]
let main argv =
    runTestsInAssembly defaultConfig argv

File: csharp/wonderland-number/test/finder-test.cs
namespace wonderland_number;

using Xunit;
using System.Collections.Generic;
using static wonderland_number.finder;

public class finder_test {

    private bool hasAllTheSameDigits(int n1, int n2) {
        var s1 = new HashSet<char>(n1.ToString().ToCharArray());
        var s2 = new HashSet<char>(n2.ToString().ToCharArray());
        return (s1 == s2);
    }

    [Fact]
    public void wonderland_number_must_have_the_following() {
        int wondernum = wonderland_number();
        Assert.Equal(6, wondernum.ToString().Length);
        Assert.True(hasAllTheSameDigits(wondernum, (wondernum * 2)));
        Assert.True(hasAllTheSameDigits(wondernum, (wondernum * 3)));
        Assert.True(hasAllTheSameDigits(wondernum, (wondernum * 4)));
        Assert.True(hasAllTheSameDigits(wondernum, (wondernum * 5)));
        Assert.True(hasAllTheSameDigits(wondernum, (wondernum * 6)));
    }
}
